hatch/src/project/mod.rs

```diff
@@ -1,81 +1,33 @@
-pub mod build;
+pub mod compiler_options;
+pub mod project_kind;
+pub mod target;
+
+pub use self::compiler_options::CompilerOptions;
+pub use self::project_kind::ProjectKind;
+pub use self::target::Target;
 
-use std::fmt;
-use std::path::PathBuf;
-use self::build::Config;
 use deps::dependency::Dependency;
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct Project {
-    name: String,
-    version: String,
-    config: Config,
-    deps: Vec<Dependency>,
-    path: PathBuf,
+  name : String,
+  version: String,
+  kind: ProjectKind,
+  #[serde(skip_serializing_if = "Option::is_none")]
+  compiler_options: Option<CompilerOptions>,
+  #[serde(skip_serializing_if = "Vec::is_empty")]
+  dependencies: Vec<Dependency>,
 }
```

Comment:
This is the serializable struct for our hatch file, that is called `Project` internally. Anything that goes into a hatch file goes here. 

TODO: dependency serialization doesn't match up as is to our defined hatch file schema.